Tests/UnitTests/Core/Propagator/CovarianceEngineTests.cpp

```diff
@@ -164,5 +171,186 @@ BOOST_AUTO_TEST_CASE(covariance_engine_test) {
   BOOST_CHECK_NE(*(std::get<0>(boundResult).covariance()),
                  Covariance::Identity());
 }
+
+BOOST_AUTO_TEST_CASE(CovarianceConversionTest) {
+  GeometryContext gctx;
+
+  std::size_t n = 100;
+
+  std::mt19937 rng{42};
+
+  std::uniform_real_distribution<double> locDist(-5., 5.);
+  std::uniform_real_distribution<double> globDist(-50., 50.);
+  std::uniform_real_distribution<double> angleDist(-2 * M_PI, 2 * M_PI);
+  std::uniform_real_distribution<double> unitDist(0, 1);
```

Comment:
`BOOST_DATA_TEST_CASE` can be very convenient for these things

Reply:
Right, but this here runs a number of tests for the different scenarios, so I'd have to add a **ton** of random numbers. This way I only need to define a handful of distributions.